### PR TITLE
Implement removals indicated by deprecations in Cantera 3.0

### PR DESCRIPTION
**Pull request by speth**
**Changes proposed in this pull request**

- Bump version to 3.1.0a1
- Remove code slated for removal and changes noted as occurring "after Cantera 3.0"
- Make relevant simplifications enabled by removal of deprecated code
- Remove some unused test data files

I tried to break up the "interesting" changes into separate commits. This might be easiest to review on a commit-by-commit basis, where the commits "Remove code deprecated in Cantera 3.0" and "Remove deprecated Python code" consist almost entirely of straight removals of deprecated code.

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: interfaces/cython/cantera/composite.py [hunk 1]
```diff
@@ -590,8 +571,8 @@ class SolutionArray(SolutionArrayBase):
         'kinetics_species_index', 'reaction', 'reactions', 'modify_reaction',
         'multiplier', 'set_multiplier', 'reaction_equations',
         'reactant_stoich_coeff', 'product_stoich_coeff',
-        'reactant_stoich_coeffs', 'product_stoich_coeffs', 'product_stoich_coeffs3',
-        'reactant_stoich_coeffs3', 'product_stoich_coeffs_reversible',
+        'reactant_stoich_coeffs', 'product_stoich_coeffs', 'product_stoich_coeffs',
```
Comment (ischoegl) [inline interfaces/cython/cantera/composite.py]: Duplicate entries.

File: src/thermo/ThermoFactory.cpp [hunk 2]
```diff
@@ -89,17 +87,11 @@ ThermoFactory::ThermoFactory()
     addDeprecatedAlias("ideal-solution-VPSS", "IdealSolnGas");
     addDeprecatedAlias("ideal-gas-VPSS", "IdealGasVPSS");
     reg("Margules", []() { return new MargulesVPSSTP(); });
-    reg("ions-from-neutral-molecule", []() { return new IonsFromNeutralVPSSTP(); });
-    addAlias("ions-from-neutral-molecule", "IonsFromNeutralMolecule");
-    addAlias("ions-from-neutral-molecule", "IonsFromNeutral");
```
Comment (ischoegl) [inline src/thermo/ThermoFactory.cpp:94]: At some point, we should start generating errors that indicate when models were removed (see #1008). Probably not something that needs to be addressed in this PR though.

**Review by ischoegl — COMMENTED** [review]
Thanks, @speth. I only have one minor thing I found (plus one comment).

File: src/thermo/ThermoFactory.cpp (hunk 2 above)
Comment (speth) [reply src/thermo/ThermoFactory.cpp:94]: No, definitely not a topic for this PR.

**Review by ischoegl — APPROVED** [review]

**Pull request merged**